presto-spi/src/main/java/com/facebook/presto/spi/block/RunLengthEncodedBlock.java

```diff
@@ -304,4 +305,66 @@ private void checkReadablePosition(int position)
             throw new IllegalArgumentException("position is not valid");
         }
     }
+
+    @Override
+    public byte getByteUnchecked(int position)
+    {
+        return ((UncheckedBlock) value).getByteUnchecked(position);
```

Comment:
Shouldn't this be `return ((UncheckedBlock) value).getByteUnchecked(0);`? In RLE, all positions have same value, right?

Reply:
The bug here wasn't that I use the position, but rather that `getOffsetBase` returned 0.  In reality, the underlying block can have its own offset, and if this method is just a passthrough, then the offset base must also be a passthrough.  I fixed it.